### PR TITLE
EZP-30626: Removed Nameable field type interface

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30626](https://jira.ez.no/browse/EZP-30626)
| **Improvement**| no
| **New feature**    | yes
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes/no
| **Doc needed**     | yes

This PR converts `eZ\Publish\SPI\FieldType` interface to abstract class. Deprecated method FieldType::getName() has new parameters (same as `getFieldName()` from `eZ\Publish\SPI\FieldType\Nameable`) and it is not deprecated anymore.  `eZ\Publish\SPI\FieldType\Nameable` interface is removed. Also other related classes are removed.

- [https://github.com/ezsystems/ezplatform-page-fieldtype/pull/91](https://github.com/ezsystems/ezplatform-page-fieldtype/pull/91)
- [https://github.com/ezsystems/ezplatform-matrix-fieldtype/pull/13](https://github.com/ezsystems/ezplatform-matrix-fieldtype/pull/13)
- [https://github.com/ezsystems/ezplatform-form-builder/pull/169](https://github.com/ezsystems/ezplatform-form-builder/pull/169)
- [https://github.com/ezsystems/ezplatform-richtext/pull/43](https://github.com/ezsystems/ezplatform-richtext/pull/43)

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Repository/Helper/NameSchemaService.php [hunk 1]
```diff
@@ -72,18 +72,18 @@ class NameSchemaService
      *
      * @param \eZ\Publish\SPI\Persistence\Content\Type\Handler $contentTypeHandler
      * @param ContentTypeDomainMapper $contentTypeDomainMapper
-     * @param NameableFieldTypeRegistry $nameableFieldTypeRegistry
+     * @param FieldTypeRegistry $fieldTypeRegistry
```
Comment (adamwojs) [inline eZ/Publish/Core/Repository/Helper/NameSchemaService.php]: Nitpick: Please use FQCN

File: eZ/Publish/Core/Repository/Helper/NameSchemaService.php [hunk 2]
```diff
@@ -58,9 +58,9 @@ class NameSchemaService
     protected $contentTypeDomainMapper;
 
     /**
-     * @var NameableFieldTypeRegistry
+     * @var fieldTypeRegistry
```
Comment (adamwojs) [inline eZ/Publish/Core/Repository/Helper/NameSchemaService.php]: Wrong docblock.

File: doc/bc/changes-8.0.md [hunk 3]
```diff
@@ -23,6 +23,11 @@ Changes affecting version compatibility with former or future versions.
   eZ Platform REST Bundle available via
   [ezsystems/ezplatform-rest](https://github.com/ezsystems/ezplatform-rest) package.
 
+* Deprecated method `getName` from the interface `eZ\Publish\SPI\FieldType\FieldType` has been changed. 
+  Now it accepts two additional parameters: `FieldDefinition $fieldDefinition` and `string $languageCode`
+  
+* Interface `eZ\Publish\SPI\FieldType\FieldType` has been transformed to abstract class.   
+
```
Comment (alongosz) [inline doc/bc/changes-8.0.md]: I'm missing information that `Nameable` has been dropped.

Comment (mikadamczyk) [reply doc/bc/changes-8.0.md]: @alongosz done

**Comment by mikadamczyk** [comment]
@alongosz it is rebased now. I think it should be merged after EE packages will be ready

**Pull request merged**